File: data/derived_cot/rq1_traces/translation/r1/py_to_cpp/code_output/ChandrasekharSieve.cpp
#include <vector>
#include <cmath>
#include <iostream>

class ChandrasekharSieve {
private:
    int n;
    std::vector<int> primes;

public:
    ChandrasekharSieve(int n) : n(n) {
        primes = generate_primes();
    }

    std::vector<int> generate_primes() {
        if (n < 2) {
            return std::vector<int>();
        }

        std::vector<bool> sieve(n+1, true);
        sieve[0] = false;
        sieve[1] = false;

        int sqrt_n = static_cast<int>(std::sqrt(n));
        for (int p = 2; p <= sqrt_n; ++p) {
            if (sieve[p]) {
                for (int i = p * p; i <= n; i += p) {
                    sieve[i] = false;
                }
            }
        }

        std::vector<int> primes_list;
        for (int i = 2; i <= n; ++i) {
            if (sieve[i]) {
                primes_list.push_back(i);
            }
        }

        return primes_list;
    }

    std::vector<int> get_primes() const {
        return primes;
    }
};

File: data/derived_cot/rq1_traces/translation/qwen/cpp_to_py/code_output/ChandrasekharSieve.py
class ChandrasekharSieve:
    def __init__(self, n):
        self.n = n
        self.primes = self._generate_primes()
    
    def _generate_primes(self):
        if self.n < 2:
            return []
        sieve = [True] * (self.n + 1)
        sieve[0] = False
        sieve[1] = False
        p = 2
        while p * p <= self.n:
            if sieve[p]:
                for i in range(p * p, self.n + 1, p):
                    sieve[i] = False
            p += 1
        primes = [i for i, is_prime in enumerate(sieve) if is_prime]
        return primes

    def get_primes(self):
        return self.primes